### DISCUSSION TITLE
如何参与到Hamster的开发中

### DISCUSSION THREAD

**Discussion by imfuxiao** [imfuxiao/Hamster#99] [Announcements]
为了方便大家参与到Hamster的开发中, 防止大家重复开发, 请按以下步骤参与:

1. 找到有标注bug/enhancement的issues, 回复下自己想要做参与这个功能/bug的开发/修复
2. 我会在将这个issue指定给您, 其他人就看到就不要在做了

## 注意事项: 

1. 提交PR时将自己的commit合并为一个. 不要提交多个commit. (对不起, 我有强迫症)
2. 提交的commit信息中请带有issues编号, 如: fix: #xx, 或者 resolved: #xxx 等等

个人喜欢用rebase合并代码, 这样看到到Log是一条直线, 比较清爽, 还请理解.

**Comment by amorphobia** [comment]
关于注意事项第一点，可否放宽一些？因为一次 commit 很难顾及周全，还会根据您的修改意见继续 commit，您在 merge 的时候可以选择 squash merge，这样 main 分支的 history 也会只有一个记录

[Squash merge](https://morningspace.github.io/tech/git-merge-stories-1/#squash-merge)

**Reply by imfuxiao** [reply]
您好，本地改好后可以使用git rebase合并相关commit

**Reply by amorphobia** [reply]
嗯嗯

**Comment by amorphobia** [comment]
在这里我也提供一种方法给其他贡献者，如何做到只有一个 commit

假设目前主仓库 main 分支的最新一个 commit hash 为 `h0`, 你的开发分支为 user_dev，而你已经 commit 过一次了，这次的 hash 记为 `h1`. 这时你需要再做一些更改，可以通过以下步骤：

1. `git reset --hard h0` —— 把当前分支同步到主仓库的 main 分支
2. `git cherry-pick -n h1` —— 将你已经 commit 的代码同步过来，但不生成新的 commit，这里的 `-n` 很重要，否则可能会自动 commit
3. 如果有冲突，先解决冲突
4. 做出新的代码修改，完成后做一次 commit
5. `git push --force` —— 强制推送到你的开发分支，如果已经有 pr 了就会自动显示在 pr 里，如果没有 pr 就可以提新的 pr 了

**Comment by kkHAIKE** [comment]
@imfuxiao 大佬。。你标记todo的算不算你当前正在做的

**Reply by imfuxiao** [reply]
对的，ToDo列是我目前开发计划内任务。

**Comment by tansongchen** [comment]
对了，建议对目前的 Issue 列表里已经做完的关闭，如 #176 #263 #277 #280 #378 。刚才想找几个有兴趣的 Issue 搞搞，发现很多其实已经完成了
